instance/models/database_server.py

```diff
@@ -120,6 +121,10 @@ class DatabaseServer(ValidateModelMixin, TimeStampedModel):
     class Meta:
         abstract = True
 
+    # Human readable identifier for database instances
+    name = models.CharField(max_length=128, unique=True)
+    description = models.CharField(max_length=128, blank=True)
```

Comment:
@bdero Replying to you here because it's easier to follow conversation in inline threads:
 
> I think this could be a good idea, but those two other name fields linked don't have a unique constraint, and if we want to add default behaviors (which one of them has) then we'd need to supply overrides anyhow. I think the uniqueness in the case of the DB models is in line with the field's purpose for this name field. I'm happy to change the length to 250 if appropriate - I just chose 128 since it's the length of the hostname field and I'm bootstrapping the new name field with the value of the hostname field in the migrations.

Yes, at the minimum changing to 250 would be good. For the uniqueness, the consistency goes both way - ie if the right choice is to have a unique name, we could add it to the other fields?

> For the general mixin refactorings & the hostname refactor, I definitely agree that it would be good to have these bread & butter mixins on a single base mixin, etc. I think for this particular piece it would be best to do this in a separate ticket + PR to avoid scope increase on this one. Maybe such a ticket could also include other refactorings we can think of to reduce redundancy. I'd be more than happy to write up and take on such an issue (it's starting to look like I might have time toward the end of the sprint). Either way I think it'd be good to have a different PR.

That would work for me - I agree that it's good to not increase the scope of this PR and risk a spillover on the task. And since this has prompted discussions, it might be good to follow-up on it on a proper ticket. I would take you on your offer to write it, thank you : )